Index: lib/components/collection-slide.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/components/collection-slide.dart b/lib/components/collection-slide.dart
new file mode 100644
--- /dev/null	(revision 3b5645b61e07fb926d8b31a71b9c4327fd2ff5a8)
+++ b/lib/components/collection-slide.dart	(revision 3b5645b61e07fb926d8b31a71b9c4327fd2ff5a8)
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_application_1/components/callapi.dart';
+import 'package:flutter_application_1/components/slider.dart';
+
+import '../screen/more.dart';
+
+class CollectionSlide extends StatelessWidget {
+  const CollectionSlide({required this.collectionData, super.key});
+
+  final SlideContent collectionData;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      child: Column(children: [
+        Container(
+            padding: EdgeInsets.only(left: 31, top: 18, right: 26, bottom: 0),
+            child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  Text(
+                    '${collectionData.nameVi}',
+                    style: const TextStyle(
+                      fontSize: 18,
+                      fontWeight: FontWeight.w700,
+                      height: 7 / 6,
+                      color: Color.fromRGBO(74, 74, 74, 1),
+                    ),
+                  ),
+                  TextButton(
+                      onPressed: () {
+                        Navigator.push(
+                            context,
+                            MaterialPageRoute(
+                                builder: (context) =>
+                                    // do something
+                                    Clickdots(dataKey: collectionData.collid)));
+                      },
+                      child: const Text(
+                        '...',
+                        style: TextStyle(
+                          fontSize: 48,
+                          height: 7 / 16,
+                          color: Color.fromRGBO(74, 74, 74, 1),
+                        ),
+                      ))
+                ])),
+        Container(
+            height: 142,
+            padding: EdgeInsets.only(left: 0, top: 2),
+            child: SliderImage1(
+              listImage: collectionData.listContent,
+              height: 142,
+              id: 'id',
+              image: 'thumnail',
+              viewportFraction: 0.3,
+            ))
+      ]),
+    );
+  }
+}
Index: lib/screen/home.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/screen/home.dart b/lib/screen/home.dart
--- a/lib/screen/home.dart	(revision 004ac157fd626d479e7dd14e8b1f2006adb7b666)
+++ b/lib/screen/home.dart	(revision 3b5645b61e07fb926d8b31a71b9c4327fd2ff5a8)
@@ -1,14 +1,10 @@
-import 'dart:convert';
-
 import 'package:carousel_slider/carousel_slider.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
-import 'package:url_launcher/url_launcher.dart';
-import 'package:http/http.dart' as http;
-import '../components/callapi.dart';
+import 'package:flutter_application_1/components/collection-slide.dart';
+
 import './detail.dart';
-import './more.dart';
-import 'package:flutter_application_1/components/slider.dart';
+import '../components/callapi.dart';
 
 class MyApp extends StatelessWidget {
   var boxshadow = [
@@ -19,6 +15,7 @@
       offset: Offset(0, 0.75), // changes position of shadow
     ),
   ];
+
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
@@ -47,13 +44,10 @@
                 dataid = data.map((item) => item.id).toList();
                 dataimagecenter = data.map((item) => item.banner).toList();
 
-                
                 return Container(
                     padding: EdgeInsets.only(
                         left: 29, right: 30, top: 18, bottom: 0),
-                    child: 
-                      
-                    CarouselSlider(
+                    child: CarouselSlider(
                       items: data
                           .map((e) => ClipRRect(
                                 child: Stack(
@@ -98,8 +92,7 @@
                         enableInfiniteScroll: true,
                         autoPlay: true,
                       ),
-                    )
-                    );
+                    ));
               } else if (snapshot.hasError) {
                 return Text('${snapshot.error}');
               }
@@ -112,7 +105,7 @@
             builder: (context, snapshot) {
               if (snapshot.hasData) {
                 List<SlideBanner> dataSlideBanner = snapshot.data!;
-                
+
                 return Container(
                     // height: 80,
                     padding: EdgeInsets.only(top: 24),
@@ -164,121 +157,11 @@
             builder: (context, snapshot) {
               if (snapshot.hasData) {
                 List<SlideContent> dataSlideContent = snapshot.data!;
-                List<String> nameVi = [];
-                List<dynamic> collid = [];
-                List<dynamic> listContent = [];
-                nameVi = dataSlideContent.map((item) => item.nameVi).toList();
-                collid = dataSlideContent.map((item) => item.collid).toList();
-                listContent =
-                    dataSlideContent.map((item) => item.listContent).toList();
-
-                return Column(children: [
-                  for (var i = 0; i < nameVi.length; i++)
-                    Container(
-                      child: Column(children: [
-                        Container(
-                            padding: EdgeInsets.only(
-                                left: 31, top: 18, right: 26, bottom: 0),
-                            child: Row(
-                                mainAxisAlignment:
-                                    MainAxisAlignment.spaceBetween,
-                                children: [
-                                  Text(
-                                    '${nameVi[i]}',
-                                    style: TextStyle(
-                                      fontSize: 18,
-                                      fontWeight: FontWeight.w700,
-                                      height: 7 / 6,
-                                      color: Color.fromRGBO(74, 74, 74, 1),
-                                    ),
-                                  ),
-                                  TextButton(
-                                      onPressed: () {
-                                        Navigator.push(
-                                            context,
-                                            MaterialPageRoute(
-                                                builder: (context) =>
-                                                    // do something
-                                                    Clickdots(
-                                                        dataKey: collid[i])));
-                                      },
-                                      child: Text(
-                                        '...',
-                                        style: TextStyle(
-                                          fontSize: 48,
-                                          height: 7 / 16,
-                                          color: Color.fromRGBO(74, 74, 74, 1),
-                                        ),
-                                      ))
-                                ])),
-                        Container(
-                            height: 142,
-                            padding: EdgeInsets.only(left: 0, top: 2),
-                            child: SliderImage1(
-                              listImage: listContent[i],
-                              height: 142,
-                              id: 'id',
-                              image: 'thumnail',
-                              viewportFraction: 0.3,
-                            )
-
-                            //  CarouselSlider(
-                            //   items: listContent[i]
-                            //       .map((e) => ClipRRect(
-                            //             child: Stack(
-                            //               children: [
-                            //                 Container(
-                            //                     padding:
-                            //                         EdgeInsets.only(bottom: 10),
-                            //                     decoration: BoxDecoration(
-                            //                       borderRadius:
-                            //                           BorderRadius.all(
-                            //                               Radius.circular(50)),
-                            //                       boxShadow: [
-                            //                         BoxShadow(
-                            //                           color: Colors.grey
-                            //                               .withOpacity(1),
-                            //                           // spreadRadius: 5,
-                            //                           blurRadius: 20,
-                            //                           offset: Offset(0,
-                            //                               0.75), // changes position of shadow
-                            //                         ),
-                            //                       ],
-                            //                     ),
-                            //                     child: InkWell(
-                            //                         onTap: () {
-                            //                           Navigator.push(
-                            //                               context,
-                            //                               MaterialPageRoute(
-                            //                                   builder:
-                            //                                       (context) =>
-                            //                                           // do something
-                            //                                           DetailPage(
-                            //                                               dataKey:
-                            //                                                   e['_id'])));
-                            //                         },
-                            //                         child: ClipRRect(
-                            //                             borderRadius:
-                            //                                 BorderRadius
-                            //                                     .circular(10),
-                            //                             child: Image.network(
-                            //                                 e['thumnail']))))
-                            //               ],
-                            //             ),
-                            //           ))
-                            //       .toList(),
-                            //   options: CarouselOptions(
-                            //     height: 142,
-                            //     viewportFraction: 0.3,
-                            //     enlargeCenterPage: false,
-                            //     enableInfiniteScroll: true,
-                            //     autoPlay: true,
-                            //   ),
-                            // ))
-                            )
-                      ]),
-                    )
-                ]);
+                return Column(
+                  children: dataSlideContent
+                      .map((data) => CollectionSlide(collectionData: data))
+                      .toList(),
+                );
               } else if (snapshot.hasError) {
                 return Text('${snapshot.error}');
               }
